### PR TITLE
Ts/ Timeline

### PR DESCRIPTION
**Pull request by lidord-wix**
Migrate Timeline to TS

### CODE REVIEW

**Comment by Inbal-Tish** [comment]
@lidord-wix ExpandableCalendar migration to TS is completed. Please resolve conflicts and see if there are changes to make.

**Comment by lidord-wix** [comment]
> @lidord-wix ExpandableCalendar migration to TS is completed. Please resolve conflicts and see if there are changes to make.

@Inbal-Tish  Done :)

File: src/timeline/Timeline.tsx [hunk 1]
```diff
@@ -2,21 +2,43 @@
...
+  eventTapped?: (event: any) => void;
+  format24h?: boolean;
+  events: {
+    start: string;
+    end: string;
+    title: string;
+    summary: string;
+    color?: string;
+  }[];
+  styles?: any;
```
Comment (Inbal-Tish) [inline src/timeline/Timeline.tsx:31]: Why 'style' is any?

File: src/timeline/Timeline.tsx [hunk 2]
```diff
@@ -2,21 +2,43 @@
...
+  scrollToFirst?: boolean;
+  renderEvent?: (event: any) => JSX.Element;
+}
+
+interface State {
+  _scrollY: any;
```
Comment (Inbal-Tish) [inline src/timeline/Timeline.tsx:37]: _scrollY should be a number

File: src/timeline/Timeline.tsx [hunk 3]
```diff
@@ -39,11 +61,14 @@ export default class Timeline extends React.PureComponent {
     events: [],
     format24h: true
   };
+  calendarHeight: number;
+  private _scrollView: any;
+  style: any;
```
Comment (Inbal-Tish) [inline src/timeline/Timeline.tsx:66]: style is not any type - please use StyleProp<ViewStyle>

File: src/timeline/Timeline.tsx [hunk 4]
```diff
@@ -39,11 +61,14 @@ export default class Timeline extends React.PureComponent {
     events: [],
     format24h: true
   };
+  calendarHeight: number;
+  private _scrollView: any;
```
Comment (Inbal-Tish) [inline src/timeline/Timeline.tsx:65]: _scrollView is a ref type not any. Also please use React.createRef and fix its usages.

File: src/timeline/Timeline.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/timeline/Timeline.tsx:31]: Please use StyleProp<ViewStyle>

File: src/timeline/Timeline.tsx [hunk 5]
```diff
@@ -2,21 +2,43 @@
...
 
 let {width: dimensionWidth} = Dimensions.get('window');
 
-export default class Timeline extends React.PureComponent {
+export interface TimelineProps {
+  start?: number;
+  end?: number;
+  eventTapped?: (event: any) => void;
+  format24h?: boolean;
+  events: {
```
Comment (Inbal-Tish) [inline src/timeline/Timeline.tsx:24]: Why no declaring type Event so we can use it elsewhere like in the packer?

File: src/timeline/Packer.ts [hunk 6]
```diff
@@ -3,11 +3,11 @@ import XDate from 'xdate';
 
 const offset = 100;
 
-function buildEvent(column, left, width, dayStart) {
-  const startTime = XDate(column.start);
-  const endTime = column.end ? XDate(column.end) : XDate(startTime).addHours(1);
+function buildEvent(column: any, left: number, width: number, dayStart: any) {
```
Comment (Inbal-Tish) [inline src/timeline/Packer.ts:6]: We need to define an Event type and use it instead of any

File: src/types.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import {ColorValue, ViewStyle, TextStyle} from 'react-native';
+import {ViewStyle, TextStyle} from 'react-native';
```
Comment (Inbal-Tish) [inline src/types.ts:1]: It doesn't look like you're using these imports

File: src/types.ts [hunk 8]
```diff
@@ -28,7 +28,7 @@ export interface Theme {
   timeLabel?: object;
   todayTextColor?: string;
   calendarBackground?: string;
-  indicatorColor?: ColorValue;
+  indicatorColor?: any;
```
Comment (Inbal-Tish) [inline src/types.ts:31]: Why changed to any?

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**